src/main/java/echoserver/EchoServer.java

```diff
@@ -5,92 +5,37 @@
 
 public class EchoServer {
     private static int port = 8080;
-    private static BufferedReader input;
-    private static PrintWriter output;
-    private static Socket clientSocket;
     private static ServerSocket serverSocket;
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         if (args.length >= 1) {
             port = Integer.parseInt(args[0]);
         } else {
             System.out.println("Port not provided. Using port 8080.");
         }
 
         try {
-            serverSocket = startServerSocket(port);
-            System.out.println("Listening for connection on port " + port);
-
-            while (true) {
-                Socket clientSocket = startClientSocket(serverSocket);
-                startSocketIO(clientSocket);
-            }
+            ServerSocketWrapper serverSocketWrapper = new ServerSocketWrapper();
+            ClientSocketWrapper clientSocketWrapper = new ClientSocketWrapper();
```

Comment:
This causes the EchoServer to never interact with the EchoClient class directly and I think that may be where I'm probably setting this up incorrectly?